Add GitHub Activity Tracker app to intern branch

Pankaj50000 opened the pull request:
“This PR adds the git_tracker_app code into the github-activity-tracker directory. The app includes a frontend (React/Vite) and backend (Node.js) for tracking GitHub repository activity, originally developed to monitor repositories and This could be useful for tracking activity in MOSIP repositories.”

Pankaj50000 committed:
Add git_tracker_app code to github-activity-tracker directory

Signed-off-by: Pankaj50000 <japankaj282@gimail.com>

ckm007 reviewed (approved):
Merging the PR for time being . Expected items in upcoming PR's:

1. github action to build frontend and backend services.
2. Readme Updated with proper documentation and architecture diagram.
3. Add a user in frontend docker.
4. details of required configurations for backend.

Commit:
Add GitHub Activity Tracker updates: github_activity.js, GitHub Actions, README, frontend Docker

Signed-off-by: Pankaj50000 <japankaj282@gimail.com>

Pankaj50000 commented:
##changes made

- Fixed DCO check by signing all commits.
- Added GitHub Actions workflows to build and deploy frontend to S3 and backend to Lambda.
- Updated README with documentation, architecture diagram, backend configurations, deployment instructions, and GitHub Actions setup.
- Deployed backend to AWS Lambda and frontend to AWS S3.
- Updated github_activity.js for AWS Lambda compatibility using aws-serverless-express.

Commit:
Update .gitignore to exclude node_modules

Signed-off-by: Pankaj50000 <japankaj282@gimail.com>

JanardhanBS-SyncByte commented on github-activity-tracker/README.md:
can you please update the readme.md put it outside and explain each part separately
